how-to/how-to-import-kata-logs-with-fluentd.md

```diff
@@ -0,0 +1,604 @@
+being stored in `/var/logs/journal`. Unfortunately, the Minikube EFK `fluentd` install stores
+most of its logs in `/run/logs`, and thus does not mount `/var/logs` into the `fluentd` pod.
+
+This can be worked around by patching the Minikube EFK addon sources to add the mount point:
+
+```patch
+iff --git a/deploy/addons/efk/fluentd-es-rc.yaml.tmpl b/deploy/addons/efk/fluentd-es-rc.yaml.tmpl
+index 75e386984..83bea48b9 100644
+--- a/deploy/addons/efk/fluentd-es-rc.yaml.tmpl
++++ b/deploy/addons/efk/fluentd-es-rc.yaml.tmpl
+@@ -44,6 +44,8 @@ spec:
+         volumeMounts:
+         - name: varlog
+           mountPath: /var/log
++        - name: runlog
++          mountPath: /run/log
+         - name: varlibdockercontainers
+           mountPath: /var/lib/docker/containers
+           readOnly: true
+@@ -57,6 +59,9 @@ spec:
+       - name: varlog
+         hostPath:
+           path: /var/log
++      - name: runlog
++        hostPath:
++          path: /run/log
+       - name: varlibdockercontainers
+         hostPath:
+           path: /var/lib/docker/containers
+```
+
+### Pull from `systemd`
+
+We will start with testing `fluentd` pulling the Kata logs directly from the system journal with the
+`fluentd` [systemd plugin](https://github.com/fluent-plugin-systemd/fluent-plugin-systemd).
+
+We modify the `fluentd` config file with the following fragment. For reference, the Minikube
+YAML can be found
+[on github](https://github.com/kubernetes/minikube/blob/master/deploy/addons/efk/fluentd-es-configmap.yaml.tmpl):
+
+> **Note:** The below `fluentd` config fragment is in the 'older style' to match  the Minikube version of
+> `fluentd`. If using a more upto date version of `fluentd`, you may need to update some parameters, such as
+> using `matches` rather than `filters` and placing `@` before `type`. Your `fluentd` should warn you in its
+> logs if such updates are necessary.
+
+```
+<source>
+  type systemd
+  tag kata-containers
+  path /run/log/journal
+  pos_file /run/log/journal/kata-journald.pos
+  filters [{"SYSLOG_IDENTIFIER": "kata-runtime"}, {"SYSLOG_IDENTIFIER": "kata-proxy"}, {"SYSLOG_IDENTIFIER": "kata-shim"}]
+  read_from_head true
+</source>
+```
+
+We then apply the new YAML, and restart the `fluentd` pod (by killing it, and letting the ReplicationController
+start a new instance, which will pick up the new ConfigurationMap):
+
+```bash
+$ kubectl apply -f new-fluentd-cm.yaml
+$ kubectl delete pod -n=kube-system fluentd-es-XXXXX
+```
+
+Now open the Kibana UI to the Minikube EFK addon, and launch a Kata QEMU based test pod in order to
+generate some Kata specific log entries:
+
+```bash
+$ minikube addons open efk
+$ cd $GOPATH/src/github.com/kata-containers/packaging/kata-deploy
+$ kubectl apply -f examples/nginx-deployment-qemu.yaml
+```
+
+Looking at the Kibana UI, we can now see that some `kata-runtime` tagged records have appeared:
+
+![Kata tags in EFK](./images/efk_kata_tag.png)
```

Comment:
Please add the sources for these binary image files so updates will be possible.

Reply:
Oops - I really should review these docs in formatted mode too :)

Reply:
Aye, file view is your friend. I also now use https://github.com/joeyespo/grip to preview the rendered markdown locally (in a browser) before I push it out to github, which makes the edit/review flow easier when working on docs for PR.